compliance_controller:
  simulate: false
  # baseline 
  #kp: [6500.0, 6885.0, 9850.0, 1400.0, 1400.0, 1400.0]
  #kd: [230.0, 645.0, 975.0, 10.0, 10.0, 10.0]

  # frictio comped max 350 lbs
  #kp: [5000.0, 5000.0, 7000.0, 300.0, 300.0, 300.0]
  #kd: [200.0, 200.0, 300.0, 10.0, 10.0, 10.0]
  #ki: [15,15,20,2,2,2]

  # max 175 lbs
  kp: [2500.0, 2500.0, 3500.0, 150.0, 150.0, 150.0]
  kd: [100.0, 100.0, 150.0, 5.0, 5.0, 5.0]
  ki: [15,15,20,2,2,2]

  # works with clipping seems P is too large
  #kp: [15000.0, 15000.0, 20500.0, 900.0, 900.0, 900.0]
  #kd: [710.0, 710.0, 950.0, 35.0, 35.0, 35.0] 
  
  # best performing with clipping
  #kp: [7500.0, 7500.0, 10250.0, 700.0, 700.0, 700.0]
  #kd: [355.0, 355.0, 475.0, 30.0, 30.0, 30.0]

  # good for .1 max error
  #kp: [15000.0, 15000.0, 20500.0, 300.0, 300.0, 300.0]
  #kd: [710.0, 710.0, 950.0, 25.0, 25.0, 25.0]
  #ki: [100.0, 100.0, 100.0, 0.00, 0.00, 0.00]

  # good for 0.01 max error
  #kp: [17000.0, 17000.0, 20000.0, 500.0, 500.0, 500.0]
  #kd: [650.0, 650.0, 750.0, 25.0, 25.0, 25.0]
  #kd: [550.0, 550.0, 650.0, 25.0, 25.0, 25.0]
  #ki: [20.0, 20.0, 20.0, 20.0, 20.0, 20.0]

  # things for integral
  use_integral_error: true
  clip_i: true
  max_trans_i: 0.05 # 5cm
  max_rot_i: 0.5 # rad (~6 deg)

  joint_state_topic: "/bravo/joint_states"
  ee_state_topic: "/bravo/ee_state"
  effort_cmd_topic: "/bravo/arm_effort_controller/command"
  toggle_cc_service_topic: "/bravo/compliance_controller/enable_compliance_controller"
  cc_cmd_topic: "/bravo/compliance_controller/command"
  ee_frame: "bravo_peg_link"

  comp_friction: true
  # max torque change per timestep
  du_max: 5.0 # n-m

  # things for the error clipping
  clip_error: true
  trans_max_error: 0.05 # m (1cm)
  rot_max_error: 0.1 # rad (~6 deg)

  # things for collision
  use_self_collision_avoidance: true
  stop_on_collision: true # will not work if use_self_collision set to false
  look_ahead_dt: 0.1

  # how much to push the robot away when collision is about to happen
  pos_repulse: 2.0
  rot_repulse: 0.0 # 1.0 = no repulsion

  # use ft_interrupt
  with_ft_sensor_interrupt: false

  # null space bias b-g
  use_nullspace: false
  nullspace_joints: [1.145, 1.586, 6.280, 1.707, 3.094, 3.157]
  nullspace_stiffness: 0.5